client/src/js/services/SubsidiesService.js

```diff
@@ -0,0 +1,87 @@
+angular.module('bhima.services')
+  .service('SubsidiesService', SubsidiesService);
+
+SubsidiesService.$inject = ['$http', 'util'];
+
+function SubsidiesService($http, util) {
+  var service = this;
+  var baseUrl = '/subsidies/';
+
+  service.read = read;
+  service.create = create;
+  service.update = update;
+  service.delete = del;
+
+  /**
+  * @desc Get an id (optionnal) and return back a list of subsidies or an subsidy
+  * @param {Integer} id, the id of the subsidy (optionnal) 
+  * @return {object} a promise object, with the response.body inside.
+  * @example
+  * service.read()
+  * .then(function (subsidies){
+  *   your code here
+  *  });
+  **/
+  function read(id) {
+    var url = baseUrl.concat(id || '');
+    return $http.get(url)
+      .then(util.unwrapHttpResponse);
+  }
+
+  /**
+  * @desc It create an subsidy
+  * @param {object} subsidy, subsidy to create 
+  * @example
+  * service.create(subsidy)
+  * .then(function (res){
+  *   your code here
+  *  });
+  **/
+  function create(subsidy) {
+    return $http.post(baseUrl, subsidy)
+      .then(util.unwrapHttpResponse);
+  }
+
+  /**
+  * @desc It updates an subsidy
+  * @param {Integer} id, subsidy id to update 
+  * @param {object} subsidy, subsidy to update 
+  * @example
+  * service.update(id, subsidy)
+  * .then(function (res){
+  *   your code here
+  *  });
+  **/
+  function update(id, subsidy) {
+    delete subsidy.id;
```

Comment:
If you are going to submit `subsidyClean` to the server, I do not think you need to delete these properties.